spring-cloud-kubernetes-config/src/main/java/org/springframework/cloud/kubernetes/config/ConfigMapPropertySource.java

```diff
@@ -196,4 +196,60 @@ else if (resourceName.equals(APPLICATION_PROPERTIES)) {
 				.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
 	}
 
+	private static ConfigMap tryRecoverConfigMap(KubernetesClient client, String name,
+			String namespace, RetryPolicy retryPolicy) throws InterruptedException {
+
+		ConfigMap map = new ConfigMap();
+		RetryPolicy normRetryPolicy = normalizeRetryPolicy(retryPolicy);
+		int attempt = 0;
+		boolean isRequestSuccessful = false;
+
+		do {
+			try {
+				map = StringUtils.isEmpty(namespace)
+						? client.configMaps().withName(name).get()
+						: client.configMaps().inNamespace(namespace).withName(name).get();
+				isRequestSuccessful = true;
+			}
+			catch (KubernetesClientException kce) {
+				String configMapMessage = "configMap with name: [" + name
+						+ "] in namespace:[" + namespace + "]";
+				LOG.warn(
+						"Can't read " + configMapMessage + ". Try " + (attempt + 1)
+								+ " of " + normRetryPolicy.getMaxAttempts() + " ...",
+						kce);
+
+				if (attempt == normRetryPolicy.getMaxAttempts() - 1) {
+					LOG.warn("Reached the maximum number of attempts to read" + " "
+							+ configMapMessage);
+					throw kce;
+				}
+				Thread.sleep(normRetryPolicy.getDelay());
```

Comment:
@geoand @iocanel @spencergibb thoughts on this?  Is there any other way we could do this?

Reply:
My first thought was to include spring-retry but I rejected it for the simple reason that not to include a new dependency. What do you think about it?

Reply:
If we don't mind adding another dependency, we have used: https://github.com/awaitility/awaitility in other projects and it's pretty great

Reply:
We have use Spring Retry in other Spring Cloud projects so I would be in favor of that